arithmetization/src/test/java/net/consensys/linea/zktracer/module/shakira/ShakiraInputsExtensiveTests.java

```diff
@@ -0,0 +1,150 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ExtendWith(UnitTestWatcher.class)
+public class ShakiraInputsExtensiveTests {
+
+  private final Random SEED = new Random(666);
+  private final List<Integer> SIZE = List.of(0, 1, 2, 8, 15, 16, 17, 31, 32, 33);
+  private final List<Integer> OFFSET =
+      List.of(0, 1, 2, 15, 16, 17, 23, 31, 32, 33, 255, 256, 257, 65535, 65535, 65537);
+  private final List<OpCode> INSTRUCTION =
+      List.of(CALL, CALLCODE, STATICCALL, DELEGATECALL, SHA3, RETURN, CREATE2);
+
+  @Tag("Weekly")
+  @ParameterizedTest
+  @MethodSource("inputs")
+  void jumpdestMasking(final int size, final int offset, final OpCode instruction) {
+    BytecodeRunner.of(
+            BytecodeCompiler.newProgram()
+                .op(CALLDATASIZE)
+                .push(0)
+                .push(0)
+                .op(CALLDATACOPY)
+                .immediate(instructionSpecificBytecode(size, offset, instruction))
+                .compile())
+        .run();
+  }
+
+  private Stream<Arguments> inputs() {
+    final List<Arguments> inputs = new ArrayList<>();
+    for (int size : SIZE) {
+      for (int offset : OFFSET) {
+        inputs.add(
+            Arguments.of(size, offset, INSTRUCTION.get(SEED.nextInt(0, INSTRUCTION.size()))));
+      }
+    }
+    return inputs.stream();
+  }
+
+  private Bytes instructionSpecificBytecode(int size, int offset, OpCode instruction) {
+    switch (instruction) {
+      case CALL, CALLCODE:
+        {
+          return BytecodeCompiler.newProgram()
+              .push(SEED.nextInt(WORD_SIZE))
+              .push(0)
+              .push(size)
+              .push(offset)
+              .push(0)
+              .push(Address.SHA256)
+              .push(1000000)
+              .op(instruction)
+              .compile();
+        }
+      case STATICCALL, DELEGATECALL:
+        {
+          return BytecodeCompiler.newProgram()
+              .push(SEED.nextInt(WORD_SIZE))
+              .push(0)
+              .push(size)
+              .push(offset)
+              .push(Address.SHA256)
+              .push(1000000)
+              .op(instruction)
+              .compile();
+        }
+      case SHA3:
+        {
+          return BytecodeCompiler.newProgram().push(size).push(offset).op(SHA3).compile();
+        }
+      case RETURN:
+        {
+          return BytecodeCompiler.newProgram()
+              .push(0)
+              .push(
+                  rightPadTo(
+                      Bytes.concatenate(
+                          Bytes.of(PUSH2.byteValue()),
+                          Bytes.ofUnsignedShort(size),
+                          Bytes.of(PUSH4.byteValue()),
+                          Bytes.ofUnsignedInt(offset),
+                          Bytes.of(RETURN.byteValue())),
+                      WORD_SIZE))
+              .op(MSTORE)
```

Comment:
As discussed on slack, wrong order of arguments. This will trigger an MXPX as the offset is
```rust
0x 61 xx 63 xx yy zz tt f3 00 .. 00
```